pastas/forecast.py

```diff
@@ -0,0 +1,300 @@
+
+    Parameters
+    ----------
+    forecasts: dict
+        Dictionary containing the forecasts data. The keys are the stressmodel names
+        and the values are lists of DataFrames containing the forecasts with a datetime
+        index and each column a time series (i.e., one ensemble member).
+
+    Returns
+    -------
+    n: int
+        The number of ensemble members in the forecasts.
+    tmin: datetime
+        The minimum datetime in the forecasts.
+    tmax: datetime
+        The maximum datetime in the forecasts.
+    index: DatetimeIndex
+        The datetime index of the forecasts.
+
+    Notes
+    -----
+    This method checks if the number of columns and indices are the same for all
+    DataFrames in the forecasts dictionary. If the number of columns or the indices are
+    not the same, a warning is printed and a ValueError is raised.
+
+    """
+    # Input validation
+    if not isinstance(forecasts, dict) or not forecasts:
+        msg = "Forecasts must be a non-empty dictionary"
+        logger.error(msg)
+        raise ValueError(msg)
+
+    n = None
+    tmax = None
+    tmin = None
+    index = None
+
+    for sm_name, fc_data in forecasts.items():
+        if not fc_data:
+            msg = f"No forecast data provided for stressmodel '{sm_name}'"
+            logger.warning(msg)
+            continue
+
+        for fc in fc_data:
+            # Check if DataFrame is empty
+            if fc.empty:
+                msg = f"Empty DataFrame in forecasts for stressmodel '{sm_name}'"
+                logger.warning(msg)
+                continue
+
+            # Check if the number of columns is the same for all DataFrames
+            if n is None:
+                n = fc.columns.size
+                tmin = fc.index[0]
+                tmax = fc.index[-1]
+                index = fc.index
+                logger.debug(f"First forecast found with {n} ensemble members")
+            # If the number of columns is not the same, raise an error
+            elif n != fc.columns.size:
+                msg = (
+                    f"The number of ensemble members is not the same for all forecasts. "
+                    f"Expected {n}, got {fc.columns.size} in stressmodel '{sm_name}'."
+                )
+                logger.error(msg)
+                raise ValueError(msg)
+            elif tmin != fc.index[0] or tmax != fc.index[-1]:
+                msg = (
+                    "The time index of the forecasts is not the same for all forecasts."
+                    "Please check the forecast data."
+                )
+                logger.error(msg)
+                raise ValueError(msg)
+
+    if n is None:
+        msg = "No valid forecast data found in any of the stressmodels"
+        logger.error(msg)
+        raise ValueError(msg)
+
+    return n, tmin, tmax, index
+
+
+def forecast(
+    ml: Model,
+    forecasts: dict[str, list[DataFrame]],
+    params: list[list[float]] | None = None,
```

Comment:
Can't we make this a DataFrame as well with columns the ensembles / iterations (same as the forecast DataFrame), index the paramter names?